src/modules/user/components/menu/userMenu.jsx

```diff
@@ -1,44 +1,431 @@
+import StarOutlineRoundedIcon from '@mui/icons-material/StarOutlineRounded';
+import AccountBoxOutlinedIcon from '@mui/icons-material/AccountBoxOutlined';
+import { useGlobalState } from 'helpers/UserPanelContext';
 
-export function UserMenu() {
-  const { t } = useTranslation(SCOPES.MODULES.USER, { useSuspense: false });
+
+export function UserMenu( props ) {
+  const [alertVisible, setAlertVisible] = useState(false);
+  const { t } = useTranslation(SCOPES.MODULES.USER, { useSuspense: false },);
+  const { t: tBar } = useTranslation(SCOPES.COMPONENTS.BAR, { useSuspense: false });
   const { user } = useContext(UserContext);
   const navigate = useNavigate();
-  const { pathname: currentRoute } = useLocation();
-  const isCurrentRoute = (route) => currentRoute === route;
-  const getStyles = (route) => [styles.item, isCurrentRoute(route) && { ...styles.selectedItem }];
+  const checkUserData = () => {
+    if (isUserDataComplete(user.data)) {
+      navigate(ROUTES.SIGNUP);
+    } else {
+      setAlertVisible(true);
+    }
+  };
+  const isMobile = useMediaQuery((theme) => theme.breakpoints.down('md'));
+  const [openCommission, setOpenCommission] = useState(false);
+  const [openEtis, setOpenEtis] = useState(false);
+  const [openLinks, setOpenLinks] = useState(false);
+  const [openInscriptions, setOpenInscriptions] = useState(false);
+  const [selectedIndex, setSelectedIndex] = useState(0);
+  const [etis, setEtis] = useState(3);
+  const [dropDownInscriptions, setdropDownInscriptions] = useState(4);
+  const [links, setLinks] = useState(5);
+  const [commission, setCommission] = useState(6);
+  const  {userData, isSignedIn} = props;
+  const userIsAdmin = userData.roles?.admin
+  const userIsSuperAdmin = userData.roles?.superadmin
+  const { toggleOpen } = useGlobalState();
+  
+  const Inscriptions = [
+    { label: t('signup'), startIndex: 9 },
+    { label: t('signupList'), startIndex: 10 },
+  ]
+
+  const Etis = [
+    { label: tBar('generalInfo'), startIndex: 11 },
+    // { label: 'Presupuesto', startIndex: 12 },  Esto se trabajara mas adelante
+    // { label: 'Inscripciones', startIndex: 13 },
+    // { label: 'Merchandising', startIndex: 14 },
+    // { label: 'Audio', startIndex: 15 },
+    { label: t('attendance'), startIndex: 16 },
+  ]
+
+  const ourLinks = [
+    { label: tBar('history'), startIndex: 17 },
+    { label: tBar('manifest'), startIndex: 18 },
+  ]
+
+  const genderCommission = [
+    { label: tBar('genderWho'), startIndex: 19 },
+    { label: tBar('genderProtocol'), startIndex: 20 },
+    { label: tBar('genderContact'), startIndex: 21 },
+  ]
+
+
+  const handleClickCommission = () => {
+    setOpenCommission(!openCommission);
+  };
+
+
+  const handleClickEtis = () => {
+    setOpenEtis(!openEtis);
+  };
+
+  const handleClickLinks = () => {
+    setOpenLinks(!openLinks);
+  };
+
+  const handleClickInscriptions = () => {
+    setOpenInscriptions(!openInscriptions);
+  };
+
+
+  const handleListItemClick = (
```

Comment:
no podemos tener 15 IFs en una función

Reply:
fijense usar algo así:

import { MenuItemIcons } from './MenuItemIcons'; // Assume this is a mapping of menu item names to their icons

export const UserMenu = () => {
  const [alertVisible, setAlertVisible] = useState(false);
  const [openSubMenus, setOpenSubMenus] = useState({});
  const { user } = useContext(UserContext);
  const navigate = useNavigate();
  const { t } = useTranslation();

  const menuItems = [
    { name: 'profile', label: t('menu.profile'), icon: <MenuItemIcons.Profile />, children: null },
    { name: 'settings', label: t('menu.settings'), icon: <MenuItemIcons.Settings />, children: [
      { name: 'privacy', label: t('menu.settings.privacy'), icon: <MenuItemIcons.Privacy /> },
      // Add other sub-menu items here
    ]},
    // Add other main menu items here
  ];

  const toggleSubMenu = (name) => {
    setOpenSubMenus(prevState => ({ ...prevState, [name]: !prevState[name] }));
  };

  const checkUserDataAndNavigate = () => {
    if (isUserDataComplete(user.data)) {
      navigate(ROUTES.SIGNUP);
    } else {
      setAlertVisible(true);
    }
  };

  const renderMenuItems = (items, parentName = '') => {
    return items.map((item) => (
      <React.Fragment key={parentName + item.name}>
        <ListItemButton onClick={() => item.children ? toggleSubMenu(item.name) : checkUserDataAndNavigate()}>
          {item.icon}
          <ListItemText primary={item.label} />
          {item.children && (openSubMenus[item.name] ? <ExpandLess /> : <ExpandMore />)}
        </ListItemButton>
        {item.children && (
          <Collapse in={openSubMenus[item.name]} timeout="auto" unmountOnExit>
            <List component="div" disablePadding>
              {renderMenuItems(item.children, `${item.name}.`)}
            </List>
          </Collapse>
        )}
      </React.Fragment>
    ));
  };

  return (
    <Box>
      <List>
        {renderMenuItems(menuItems)}
      </List>
      {alertVisible && /* Render your alert component here */}
    </Box>
  );
};

Reply:
Se utilizó un enfoque similar al compartido, implementando un array de objetos para los botones del menú